cloudpathlib/gs/gsclient.py

```diff
@@ -196,17 +196,22 @@ def _move_file(self, src: GSPath, dst: GSPath, remove_src: bool = True) -> GSPat
 
         return dst
 
-    def _remove(self, cloud_path: GSPath) -> None:
-        if self._is_file_or_dir(cloud_path) == "dir":
+    def _remove(self, cloud_path: GSPath, missing_ok: bool = True) -> None:
+        file_or_dir = self._is_file_or_dir(cloud_path)
+        if file_or_dir == "dir":
             blobs = [
                 b.blob for b, is_dir in self._list_dir(cloud_path, recursive=True) if not is_dir
             ]
             bucket = self.client.bucket(cloud_path.bucket)
             for blob in blobs:
                 bucket.get_blob(blob).delete()
-        elif self._is_file_or_dir(cloud_path) == "file":
+        elif file_or_dir == "file":
             bucket = self.client.bucket(cloud_path.bucket)
             bucket.get_blob(cloud_path.blob).delete()
+        else:
+            # Does not exist
+            if not missing_ok:
+                raise FileNotFoundError(f'File does not exist: {cloud_path}')
```

Comment:
I hesitate to do it generically because it would add an unnecessary call to `self.client`: one call to check for existence, another to remove, whereas this way only one call is needed.
This may be unnecessary optimization though. If you want me to add an `if not self.exists()` to the generic method I'm OK with that

Reply:
Let's leave it as you implemented for now. Eventually we may end up caching metadata like file_or_dir/existence, but for the time being it's nice to just to have the one check.